topi/python/topi/nn/sparse.py

```diff
@@ -101,3 +101,106 @@ def _compute_block(i, nb_j, j):
         (m, num_blocks * bs_r),
         lambda m, n: bsrmm_block[m, n // bs_r, n % bs_r],
         tag="sparse_dense_bsrmm")
+
+@tvm.target.generic_func
+def sparse_transpose(sparse_data, sparse_indices, sparse_indptr):
+    """
+    Transpose a square sparse matrix,
+    `A` is an n-by-n sparse matrix in the CSR format.
+    ** Currently only support Square Matrices **
+
+    Parameters
+    ----------
+    sparse_data : tvm.Tensor
+        1-D with shape [nonzeros], dtype of 'float32'
```

Comment:
Does the data tensor have to be of type `float32` or can we play around with other datatypes?

Reply:
Yes. We can support more data types in the future.